sdk/cosmos/azure-cosmos/README.md

```diff
@@ -1,67 +1,310 @@
+
+Perform parameterized queries by passing a dictionary containing the parameters and their values to [Container.query_items][ref_container_query_items]:
+
+```Python
+discontinued_items = container.query_items(
+    query='SELECT * FROM products p WHERE p.productModel = @model',
+    parameters=[
+        dict(name='@model', value='Model 7')
+    ],
+    enable_cross_partition_query=True
+)
+for item in discontinued_items:
+    print(json.dumps(item, indent=True))
+```
+
+For more information on querying Cosmos DB databases using the SQL API, see [Query Azure Cosmos DB data with SQL queries][cosmos_sql_queries].
+
+### Get database properties
+
+Get and display the properties of a database:
+
+```Python
+database = client.get_database_client(database_name)
+properties = database.read()
+print(json.dumps(properties))
+```
+
+### Modify container properties
+
+Certain properties of an existing container can be modified. This example sets the default time to live (TTL) for items in the container to 10 seconds:
+
+```Python
+database = client.get_database_client(database_name)
+container = database.get_container_client(container_name)
+database.replace_container(container,
+                           partition_key=PartitionKey(path="/productName"),
+                           default_ttl=10,
+                           )
+# Display the new TTL setting for the container
+container_props = container.read()
+print(json.dumps(container_props['defaultTtl']))
 ```
 
-Most of the test files under test sub-folder require you to enter your Azure Cosmos master key and host endpoint: 
+For more information on TTL, see [Time to Live for Azure Cosmos DB data][cosmos_ttl].
+
+## Troubleshooting
+
+### General
 
-    masterKey = '[YOUR_KEY_HERE]'
-    host = '[YOUR_ENDPOINT_HERE]'
+When you interact with Cosmos DB using the Python SDK, errors returned by the service correspond to the same HTTP status codes returned for REST API requests:
 
-To run the tests:
+[HTTP Status Codes for Azure Cosmos DB][cosmos_http_status_codes]
 
-    $ python -m unittest discover -s .\test -p "*.py" 
+For example, if you try to create a container using an ID (name) that's already in use in your Cosmos DB database, a `409` error is returned, indicating the conflict. In the following snippet, the error is handled gracefully by catching the exception and displaying additional information about the error.
 
-    If you use Microsoft Visual Studio, open the project file python.pyproj,
-    and run all the tests in Test Explorer.
+```Python
+try:
+    database.create_container(id=container_name, partition_key=PartitionKey(path="/productName")
+except HTTPFailure as e:
+    if e.status_code == 409:
+        print("""Error creating container.
+HTTP status code 409: The ID (name) provided for the container is already in use.
+The container name must be unique within the database.""")
+    else:
+        raise
+```
```

Comment:
Might be worth mentioning (here, or elsewhere) that the `get_X_client` methods will never fail with an exception since that seemed to trip up the UX study participants some.